### DISCUSSION TITLE
[Feature] Integrate E2E with existing k8s cluster

### DISCUSSION THREAD

**Discussion by qqeasonchen** [apache/skywalking#8051] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Mostly private enviorment already has k8s clusters, it would be nice to provide an easy way to integrate E2E with existing k8s, no need to rely on kind, thanks.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
This was discussed, the main idea we didn't provide this is for making sure the system env consistent. K8s is not always same or sharing the same APIs or formats of deployment files from version to version.
Exposing the process into your private(a random version or tool) would be very risky.

Could you share why you want this?

**Reply by qqeasonchen** [reply]
Thanks for your reply, we have our own k8s-installer to build enviroment, so it would be heavy for us to migarate to kind and re-install our k8s.

**Reply by kezhenxu94** [reply]
I don't remember we discussed this. This is necessary when the testing environment uses more resources than what GitHub / KinD can offer. Also when users have more powerful Kubernetes engine and want to speed up the tests it is needed.

**Reply by kezhenxu94** [reply]
Allowing to use existing env doesn't mean the env will be reused for multiple tests cases, users can spin up a new cluster (e.g. GKE) and use it only for that workflow and then tear down after the workflow. This is users choice but we need to provide the possibility.

**Reply by wu-sheng** [reply]
> I don't remember we discussed this.

There was an unintentional discussion(I asked) about why infra-e2e should include KinD. Anyway.

___

Currently, we just finished the infra-e2e in the open source offering GHA, which is the major purpose for SkyWalking itself. I am OK if you want to expand the landscape to more commercial or private CI platform(or vendor sponsored).
But, version mis-match still could be reported to us, we need to be clear in the document about the scopes such as what are bundled.
